examples/gpio/config.c

```diff
@@ -0,0 +1,248 @@
+    if (rc != TPM_RC_SUCCESS) {
+        printf("wolfTPM2_Init failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+        goto exit;
+    }
+    printf("wolfTPM2_Init: success\n");
+
+    /* Get TPM capabilities, to discover the TPM vendor */
+    rc = wolfTPM2_GetCapabilities(&dev, &caps);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("wolfTPM2_GetCapabilities failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+    }
+
+    /* Enable TPM2_GPIO_Config command */
+    if (caps.mfg != TPM_MFG_STM) {
+        printf("Extra GPIO is supported only on ST33 TPM 2.0 modules\n");
+        goto exit;
+    }
+
+#ifdef DEBUG_WOLFTPM
+    printf("Trying to remove NV index 0x%8.8X used for GPIO\n", nvIndex);
+#endif
+    /* Make sure NV Index for this GPIO is cleared before use
+     * This way we make sure a new GPIO config can be set
+     */
+    rc = wolfTPM2_NVDelete(&dev, TPM_RH_OWNER, nvIndex);
+#ifdef DEBUG_WOLFTPM
+    if (rc == TPM_RC_SUCCESS) {
+        printf("NV index undefined\n");
+    }
+    else {
+        printf("wolfTPM2_NVDelete failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+    }
+#endif
+
+
+    XMEMSET(&setCmdSet, 0, sizeof(setCmdSet));
+    setCmdSet.authHandle = TPM_RH_PLATFORM;
+    setCmdSet.commandCode = TPM_CC_GPIO_Config;
+    setCmdSet.enableFlag = 1;
+    rc = TPM2_SetCommandSet(&setCmdSet);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("TPM2_SetCommandSet failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+        goto exit;
+    }
+
+    /* Configuring GPIO */
+    XMEMSET(&gpio, 0, sizeof(gpio));
+    gpio.authHandle = TPM_RH_PLATFORM;
+    gpio.config.count = 1;
+    gpio.config.gpio[0].name = gpioNum;
+    gpio.config.gpio[0].mode = gpioMode;
+    gpio.config.gpio[0].index = nvIndex;
+    printf("Trying to configure GPIO%d...\n", gpio.config.gpio[0].name);
+    rc = TPM2_GPIO_Config(&gpio);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("TPM2_GPIO_Config failed 0x%x: %s\n", rc,
+            TPM2_GetRCString(rc));
+        goto exit;
+    }
+    printf("TPM2_GPIO_Config success\n");
+
+    /* Configure NV Index for access to this GPIO */
+    XMEMSET(&nv, 0, sizeof(nv));
+    XMEMSET(&parent, 0, sizeof(parent));
+    /* Prep NV attributes */
+    parent.hndl = TPM_RH_OWNER;
+    rc = wolfTPM2_GetNvAttributesTemplate(parent.hndl, &nvAttributes);
+    if (rc != 0) {
+        printf("Setting NV attributes failed\n");
+        goto exit;
+    }
+    /* Define NV Index for GPIO */
+    rc = wolfTPM2_NVCreateAuth(&dev, &parent, &nv, nvIndex, nvAttributes,
+                               sizeof(BYTE), (byte*)gNvAuth, sizeof(gNvAuth)-1);
+    if (rc != 0 && rc != TPM_RC_NV_DEFINED) {
+        printf("Creating NV Index for GPIO acccess failed\n");
+        goto exit;
+    }
+    printf("NV Index for GPIO access created\n");
+
+    /* GPIO configured as an input, requires an extra configuration step */
+    if (gpioInput) {
+        rc = wolfTPM2_NVWriteAuth(&dev, &nv, nvIndex, &dummy, sizeof(dummy), 0);
```

Comment:
Please add note that "dummy" value is discarded for GPIO and is required for the NV read to work on GPIO.